utils/src/main/scala/za/co/absa/enceladus/utils/implicits/StructFieldImplicits.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.implicits
+
+import org.apache.spark.sql.types.StructField
+import za.co.absa.enceladus.utils.types.Defaults
+import scala.util.Try
+
+object StructFieldImplicits {
+  implicit class StructFieldImprovements(val structField: StructField) {
+    def getMetadata( key: String): Option[String] = {
+      Try(structField.metadata.getString(key)).toOption
+    }
+
+    def readPatternInfo(): PatternInfo = {
+      val patternOpt: Option[String] = getMetadata("pattern")
+      PatternInfo(patternOpt.getOrElse(Defaults.getGlobalFormat(structField.dataType)), patternOpt.isEmpty)
+    }
+  }
+
+  case class PatternInfo (pattern: String, isDefault: Boolean)
```

Comment:
I would expect case class on top

Reply:
I prefer putting `implicit` stuff above all else. 😄

Reply:
I get that, but the case class is on the level with the object.